sdk/core/core-client/src/interfaces.ts

```diff
@@ -358,14 +358,48 @@ export type RawResponseCallback = (
 
 /**
  * Used to map raw response objects to final shapes.
- * Mostly useful for unpacking/packing Dates and other encoded types that
- * are not intrinsic to JSON.
+ * Helps packing and unpacking Dates and other encoded types that are not intrinsic to JSON.
+ * Also allows pulling values from headers, as well as inserting default values and constants.
  */
 export interface Serializer {
+  /**
+   * The provided model mapper.
+   */
   readonly modelMappers: { [key: string]: any };
+  /**
+   * Whether the contents are XML or not.
+   */
   readonly isXML: boolean;
+
+  /**
+   * Validates constraints, if any. This function will throw if the provided value does not match those constraints.
+   * @param mapper - The definition of data models.
+   * @param value - The value.
+   * @param objectName - Name of the object. Used in the error messages.
+   */
   validateConstraints(mapper: Mapper, value: any, objectName: string): void;
+
+  /**
+   * Serialize the given object based on its metadata defined in the mapper.
+   * Deserialize the given object based on its metadata defined in the mapper.
```

Comment:
I think the second line was added by mistake?
```suggestion
   * Serialize the given object based on its metadata defined in the mapper.
```

Reply:
thank you!